backend/webserver/wittymail_server/emailapi_broker.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import openpyxl
+import emailapi
+import re
+import util.logger as logger
+
+log = logger.get_logger(__name__)
+
+# Names of the fodder to be used to create the emails
+email_fodder_names = []
+# Fodder content used to create the emails
+email_fodder = []
+
+# Index of the emailid in the email_fodder_names[]. The emailid will be used in the 'to' field
+EMAIL_FODDER_TO_INDEX = 0 
+# Index of the emailid in the email_fodder_names[]. The emailid will be used in the 'cc' field
+EMAIL_FODDER_CC_INDEX = 0
+
+# Extended fodder names that will be appended to the fodder names provided by the user (i.e. email_fodder_names)
+extended_email_fodder_names = ['email_subject', 'email_body', 'email_attachment', 'email_sent']
+extended_email_fodder_names_EMAIL_BODY_INDEX = -3
+extended_email_fodder_names_EMAIL_SUBJECT_INDEX = -4
+extended_email_fodder_names_EMAIL_ATTACHMENT_INDEX = -2
+
+# Extended fodder that will be appended to the fodder provided by the user (i.e. email_fodder[])
+extended_default_email_fodder = [None, None, None, False]
+
+# TODO Add check to detect extention type
+def save_fodder_from_file(loc, email_fodder_names_template = None):
+    global email_fodder
+    global email_fodder_names
+   
+    _email_fodder = []  
+
+    # TODO Check if we should read wb_obj.active or wb_obj[0]
+    wb_obj = openpyxl.load_workbook(loc)   
+    sheet_obj = wb_obj.active
+
+    for i in range(1, sheet_obj.max_row + 1):
+        cells = []
+
+        for j in range(1, sheet_obj.max_column + 1):
+            cells.append(str(sheet_obj.cell(row = i, column = j).value).encode('UTF8'))
+        _email_fodder.append(cells)
+
+    assert email_fodder_names_template == None or email_fodder_names == email_fodder_names_template, \
+           "**** The headers in the excel sheet seems to be invalid ****"
```

Comment:
I'm assuming these asserts raise an AssertError exception so that Flask will return a HTTP 500 to the REST API and I can display an 'Internal Error, call Ajay Nair and curse him' message, right?

Reply:
Haha Yes